### PR TITLE
Replacing the old concept of 'patch' by 'pull request' or 'contribution'

### PR DESCRIPTION
**Pull request by avallbona**
Replacing the old concept of 'patch' by 'pull request' or 'contribution'

This work is part of effort done in the Sprints in DjangoCon Europe in Vigo with Daniele Procida @evildmp

Task picked from the task list in https://docs.google.com/spreadsheets/d/16UTGwtAoOwznc46cszbwAHU9xbukXnnpwG-faE94Rw8/edit?gid=0#gid=0

Discussed this subject with Natalia @nessita

### CODE REVIEW

File: docs/intro/contributing.txt [hunk 1]
```diff
@@ -398,7 +398,7 @@ function to the correct file.
 Running Django's test suite for the second time
 ===============================================
 
-Once you've verified that your patch and your test are working correctly, it's
+Once you've verified that your chnages and your test are working correctly, it's
```
Comment (felixxm) [inline docs/intro/contributing.txt:401]: Typo:
```suggestion
Once you've verified that your changes and your test are working correctly, it's
```

**Review by evildmp — APPROVED** [review]
I think this is excellent, especially in that the changes it introduces are technology-agnostic where appropriate (e.g. it speaks of _changes_ and _contributions_ rather than _pull requests_ except where needed).

File: docs/intro/contributing.txt (hunk 1 above)
Comment (avallbona) [reply docs/intro/contributing.txt]: @felixxm addressed in c525981b267f6fd7611d300e2df7666ca7d9ca4b

File: docs/intro/contributing.txt [hunk 2]
```diff
@@ -91,7 +91,7 @@ Installing Git
 ==============
 
 For this tutorial, you'll need Git installed to download the current
-development version of Django and to generate patch files for the changes you
+development version of Django and to generate a pull request for the changes you
```
Comment (nessita) [inline docs/intro/contributing.txt:94]: This looks great but it needs to be wrapped at 79cols. Could you please adjust this sentence with that in mind?

File: docs/intro/contributing.txt [hunk 3]
```diff
@@ -398,7 +398,7 @@ function to the correct file.
 Running Django's test suite for the second time
 ===============================================
 
-Once you've verified that your patch and your test are working correctly, it's
+Once you've verified that your changes and your test are working correctly, it's
```
Comment (nessita) [inline docs/intro/contributing.txt:401]: Same here! :-)

File: docs/intro/contributing.txt [hunk 4]
```diff
@@ -450,8 +450,8 @@ preview the HTML that will be generated.
 Previewing your changes
 =======================
 
-Now it's time to go through all the changes made in our patch. To stage all the
-changes ready for commit, run:
+Now it's time to go through all the changes made in our contribution. To stage
```
Comment (nessita) [inline docs/intro/contributing.txt:453]: I would suggest to use `branch` here...
```suggestion
Now it's time to go through all the changes made in our branch. To stage
```

File: docs/intro/contributing.txt [hunk 5]
```diff
@@ -578,14 +577,14 @@ codebase.
 More information for new contributors
 -------------------------------------
 
-Before you get too into writing patches for Django, there's a little more
+Before you get too into writing contributions for Django, there's a little more
```
Comment (nessita) [inline docs/intro/contributing.txt:580]: Maybe?
```suggestion
Before you get too into contributing to Django, there's a little more
```

File: docs/intro/contributing.txt [hunk 6]
```diff
@@ -578,14 +577,14 @@ codebase.
...
 information on contributing that you should probably take a look at:
 
 * You should make sure to read Django's documentation on
   :doc:`claiming tickets and submitting patches
   </internals/contributing/writing-code/submitting-patches>`.
   It covers Trac etiquette, how to claim tickets for yourself, expected
-  coding style for patches, and many other important details.
+  coding style for contributions, and many other important details.
```
Comment (nessita) [inline docs/intro/contributing.txt:587]: I would just omit it here:
```suggestion
  coding style (both for code and docs), and many other important details.
```

File: docs/intro/contributing.txt [hunk 7]
```diff
@@ -600,11 +599,11 @@ Finding your first real ticket
 ------------------------------
 
 Once you've looked through some of that information, you'll be ready to go out
-and find a ticket of your own to write a patch for. Pay special attention to
-tickets with the "easy pickings" criterion. These tickets are often much
+and find a ticket of your own to write a contribution for. Pay special attention
```
Comment (nessita) [inline docs/intro/contributing.txt:602]: Maybe?
```suggestion
and find a ticket of your own to contribute to. Pay special attention
```

File: docs/intro/contributing.txt [hunk 8]
```diff
@@ -600,11 +599,11 @@ Finding your first real ticket
...
+to tickets with the "easy pickings" criterion. These tickets are often much
 simpler in nature and are great for first time contributors. Once you're
-familiar with contributing to Django, you can move on to writing patches for
-more difficult and complicated tickets.
+familiar with contributing to Django, you can move on to writing contributions
```
Comment (nessita) [inline docs/intro/contributing.txt:605]:
```suggestion
familiar with contributing to Django, you can move on to working on
```

File: docs/intro/contributing.txt [hunk 9]
```diff
@@ -626,5 +625,6 @@ After a ticket has a patch, it needs to be reviewed by a second set of eyes.
 After submitting a pull request, update the ticket metadata by setting the
 flags on the ticket to say "has patch", "doesn't need tests", etc, so others
 can find it for review. Contributing doesn't necessarily always mean writing a
-patch from scratch. Reviewing existing patches is also a very helpful
-contribution. See :doc:`/internals/contributing/triaging-tickets` for details.
+contribution from scratch. Reviewing existing contributions is also a very
```
Comment (nessita) [inline docs/intro/contributing.txt:628]:
```suggestion
code from scratch. Reviewing existing contributions is also a very
```

**Review by nessita — CHANGES_REQUESTED** [review]
Thank you, this looks great! :star2:
I proposed a few tweaks, let me know what you think!

**Comment by avallbona** [comment]
@nessita I agree with your corrections, I addressed all of them in eccf56b7f15e2a315a61ff013132388850cdd79d

**Review by nessita — APPROVED** [review]
Thank you @avallbona for the changes! I submitted a few extra tweaks. Will merge when the CI is green.

**Comment by avallbona** [comment]
@nessita as dicussed, I'm extending a bit the scope fixing also this page: https://docs.djangoproject.com/en/5.0/internals/contributing/writing-code/submitting-patches/

**Pull request merged**